app/src/main/java/com/example/smsrc/commands/views/fragments/PlaySoundFragment.java

```diff
@@ -0,0 +1,68 @@
+package com.example.smsrc.commands.views.fragments;
+
+import android.os.Bundle;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.fragment.app.Fragment;
+import androidx.navigation.NavController;
+import androidx.navigation.Navigation;
+
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.Button;
+
+import com.example.smsrc.MainActivity;
+import com.example.smsrc.R;
+import com.example.smsrc.commands.model.CommandsContract;
+import com.example.smsrc.permissions.utils.Crypto;
+import com.example.smsrc.sms.model.SMS;
+import com.example.smsrc.sms.presenter.SMSPresenter;
+import com.google.android.material.textfield.TextInputEditText;
+
+public class PlaySoundFragment extends Fragment {
+
+    private NavController navController;
+    private SMSPresenter smsPresenter;
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container,
+                             Bundle savedInstanceState) {
+        return inflater.inflate(R.layout.fragment_play_sound, container, false);
+    }
+
+    @Override
+    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
+        super.onViewCreated(view, savedInstanceState);
+        navController = Navigation.findNavController(view);
+        smsPresenter = new SMSPresenter(getContext());
+        MainActivity mainActivity = (MainActivity)this.getActivity();
+        mainActivity.hideNavigation();
+        initListeners(view);
+    }
+
+    private void initListeners(View view) {
+        Button sendButton = view.findViewById(R.id.play_sound_send);
+        sendButton.setOnClickListener(e->{
+            sendSms(view);
+            navController.popBackStack();
+        });
+    }
+
+    private void sendSms(View view) {
+        String phoneNumber = ((TextInputEditText)view.findViewById(R.id.play_sound_number)).getText().toString();
+        String username = ((TextInputEditText)view.findViewById(R.id.play_sound_username)).getText().toString();
+        String password = ((TextInputEditText)view.findViewById(R.id.play_sound_password)).getText().toString();
+
+        String randomness = Crypto.generateRandomness();
+        String credentials = Crypto.encrypt(username + password, randomness);
```

Comment:
in the executor it was username + randomness + password

Reply:
password is not encrypted before send and it is not available at reciever

Reply:
why not use the current user name and password instead of sending one this could be a guest of level 2 account